File: 2154.keep-multiplying-found-values-by-two/solution.go
func findFinalValue(nums []int, original int) int {
    v := make([]bool, 1001)
    for _, n := range nums {
        v[n] = true
    }
    n := original
    for n < len(v) && v[n] {
        n += n
    }
    return n
}

File: 2154.keep-multiplying-found-values-by-two/solution.kt
class Solution {
    fun findFinalValue(nums: IntArray, original: Int): Int {
        val v = BooleanArray(1001)
        nums.forEach { v[it] = true }
        var n = original
        while (n < v.size && v[n]) {
            n += n
        }
        return n
    }
}